core/src/main/java/com/sequenceiq/cloudbreak/core/flow2/cluster/ccm/upgrade/CcmUpgradeEvent.java

```diff
@@ -0,0 +1,35 @@
+package com.sequenceiq.cloudbreak.core.flow2.cluster.ccm.upgrade;
+
+import com.sequenceiq.cloudbreak.reactor.api.event.cluster.upgrade.ccm.CcmRemoveAutoSshResult;
+import com.sequenceiq.cloudbreak.reactor.api.event.cluster.upgrade.ccm.CcmReregisterToClusterProxyResult;
+import com.sequenceiq.cloudbreak.reactor.api.event.cluster.upgrade.ccm.CcmUnregisterHostsResult;
+import com.sequenceiq.cloudbreak.reactor.api.event.cluster.upgrade.ccm.CcmUpgradeFailed;
+import com.sequenceiq.cloudbreak.reactor.api.event.cluster.upgrade.ccm.CcmUpgradePreparationFailed;
+import com.sequenceiq.cloudbreak.reactor.api.event.cluster.upgrade.ccm.CcmUpgradePreparationResult;
+import com.sequenceiq.flow.core.FlowEvent;
+import com.sequenceiq.flow.event.EventSelectorUtil;
+
+public enum CcmUpgradeEvent implements FlowEvent {
+    CCM_UPGRADE_EVENT("CCM_UPGRADE_TRIGGER_EVENT"),
+    CCM_UPGRADE_PREPARATION_FINISHED_EVENT(EventSelectorUtil.selector(CcmUpgradePreparationResult.class)),
+    CCM_UPGRADE_PREPARATION_FAILED_EVENT(EventSelectorUtil.selector(CcmUpgradePreparationFailed.class)),
```

Comment:
Is it justified to have a separate failure event for this scenario, instead of the general `CCM_UPGRADE_FAILURE_EVENT`?

Reply:
We will see when fill the flow with actual code